src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -149,13 +149,29 @@ public async Task<T> GetConfigurationAsync()
         /// Obtains an updated version of Configuration.
         /// </summary>
         /// <param name="cancel">CancellationToken</param>
-        /// <returns>Configuration of type T.</returns>
-        /// <remarks>If the time since the last call is less than <see cref="BaseConfigurationManager.AutomaticRefreshInterval"/> then <see cref="IConfigurationRetriever{T}.GetConfigurationAsync"/> is not called and the current Configuration is returned.</remarks>
+        /// <returns>Configuration of type <typeparamref name="T"/>.</returns>
+        /// <remarks>
+        /// If the time since the last call is less than <see cref="BaseConfigurationManager.AutomaticRefreshInterval"/>
+        /// then <see cref="IConfigurationRetriever{T}.GetConfigurationAsync"/> is not called and the current Configuration is returned.
+        /// By default, this method blocks until the configuration is retrieved the first time. After the configuration is retrieved,
+        /// updates will happen in the background. Failures to retrieve the configuration on the background thread will be logged.
+        /// If this operation is configured to be blocking through the switch 'Switch.Microsoft.IdentityModel.UpdateConfigAsBlocking'
+        /// then this method will block when the configuration needs to be updated or hasn't been retrieved. If the configuration
+        /// is unable to be retrieved, an exception will be thrown.
+        /// </remarks>
```

Comment:
Did I capture the trade-off correctly, @keegan-caruso?

```suggestion
        /// By using the app context switch you choose what works best for you when there is a signing key update:
        /// either have validation failures until the new metadata is retrieved (non-blocking behavior), or block all requests
        /// until the new metadata is retrieved and process them at that time. The latest case can lead to thread starvation 
        /// if your service receives a lot of concurrent requests and the signing keys are rotated.
        /// </remarks>
```